backend/controllers/prisonerProfile/prisonerAdjudicationDetails.js

```diff
@@ -0,0 +1,59 @@
+const { serviceUnavailableMessage } = require('../../common-messages')
+const { formatTimestampToDate, formatTimestampToDateTime, putLastNameFirst } = require('../../utils')
+const {
+  app: { notmEndpointUrl: dpsUrl },
+} = require('../../config')
+
+module.exports = ({ elite2Api, logError }) => async (req, res) => {
+  const { offenderNo, adjudicationNumber } = req.params
+
+  try {
+    const [prisonerDetails, adjudicationDetails] = await Promise.all([
+      elite2Api.getDetails(res.locals, offenderNo),
+      elite2Api.getAdjudicationDetails(res.locals, offenderNo, adjudicationNumber),
+    ])
+
+    const { firstName, lastName } = prisonerDetails
+    const { incidentTime, reportTime, hearings, reporterFirstName, reporterLastName } = adjudicationDetails
+
+    return res.render('prisonerProfile/prisonerAdjudicationDetails.njk', {
+      adjudicationDetails: {
+        ...adjudicationDetails,
+        hearings: hearings.map(hearing => {
+          const { hearingTime, heardByFirstName, heardByLastName, results } = hearing
+          const comments = []
```

Comment:
I wonder if it'll be easier to follow if you mapped all the results first, then pulled the comments out after?

So like:

`const results = results.map(result => ({
              ...result,
              sanctions: result.sanctions.map(sanction => {
                const { comment, effectiveDate, statusDate } = sanction
                if (comment) comments.push(comment)

                return {
                  ...sanction,
                  effectiveDate: effectiveDate && formatTimestampToDate(effectiveDate),
                  statusDate: statusDate && formatTimestampToDate(statusDate),
                }
              })
       const comments = results.flatMap(result => result.sanctions)
.filter(sanction => santion.comment)
.map(sanction.comment)
`

Reply:
Funnily enough I did do that first and then changed to what I've got now so it was mapping through less. I'm easy either way though.

Reply:
It seemed a bit out of place in our code base and had to chase it about it to fully understand it.